File: backend/models/Notification.js
const mongoose = require('mongoose');

const notificationSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  notificationType: {
    type: String,
    required: true,
    // e.g. 'technical_interview', 'coding', 'resume', 'hr_interview',
    //      'placement_recommendation', 'aptitude', 'login_summary'
  },
  title: {
    type: String,
    required: true
  },
  message: {
    type: String,
    default: ''
  },
  status: {
    type: String,
    enum: ['unread', 'read'],
    default: 'unread'
  },
  emailSent: {
    type: Boolean,
    default: false
  }
}, { timestamps: true });

module.exports = mongoose.model(
  'Notification',
  notificationSchema,
  'Notifications'
);

File: backend/models/notification.py
"""models/notification.py — Notification table (replaces Mongoose Notification.js)"""
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Text, Enum
from sqlalchemy.sql import func
from database import Base


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    notification_type = Column(String(100), nullable=False)
    title = Column(String(500), nullable=False)
    message = Column(Text, default="")
    status = Column(Enum("unread", "read"), default="unread")
    email_sent = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
